docs/testing.md

```diff
@@ -0,0 +1,179 @@
+---
+id: testing
+title: Testing
+---
+
+As your codebase expands, small errors and edge cases you don’t expect can cascade into larger failures. Bugs lead to bad user experience and ultimately, business losses. One way to prevent fragile programming is to test your code before releasing it into the wild. In this guide, we will cover different ways to ensure your app works as expected, ranging from static analysis to End to End tests.
+
+## Static Analysis
+
+The first step to improve your code quality is to start using static analysis tools. Static analysis checks your code for errors as you write it, but without running any of that code.
+
+- **Linters** analyze code to catch common errors, such as unused code, help avoid pitfalls or flag style guide no-nos like using tabs instead of spaces (or vice versa, depending on your configuration). Linters are configured by a set of rules - and your code needs to follow them.
+- **Type checking** ensures that the construct you’re passing to a function matches what the function was designed to accept, preventing passing a string to a counting function that expects an integer, for instance.
+
+React Native comes with two such tools configured out of the box: [ESLint](https://eslint.org/) for linting and [Flow](https://flow.org/en/docs/) for type checking. You can also use [TypeScript](https://www.typescriptlang.org/), which is a typed language that compiles to plain JavaScript.
+
+## Writing Testable Code
+
+To start with tests, you first need to write code that is testable. Consider a car’s engine. For some reason it won’t start. In older cars, figuring out what is wrong with the engine requires the mechanic to take the engine apart, piece by piece, until they can figure out the issue. In modern cars, engines are outfitted with sensors that keep an eye on parts of the system, determining the exact location of part failures—sometimes before the engine even breaks down! Sensors changed the way engines were designed and manufactured.
```

Comment:
I wrote this bit ;p Often times looping in a technology folks already know and are familiar with to paint a picture makes the information more sticky.

Reply:
I like that we're using a metaphor to explain tests!

I think there's a difference between the way sensors in cars work and the way code tests work because sensors do not run parts of the engine like tests do. Sensors are more like static analysis in the above section. Is there a better metaphor for this section we could use?

Reply:
How would this sound?

To start with tests, you first need to write code that is testable. Consider an aircraft manufacturing process - before a model first takes off, individual parts are tested to guarantee they are safe and function correctly. For example, the durability of wings is tested by bending them under extreme load, engine parts are tested for their durability, windshield is tested against simulated bird impact, and much more.

Reply:
I endorse this new metaphor! Airplanes are pretty cool, too!

Reply:
ok, I'm considering this resolved then